landlab/components/resource_redistribution/resource_redistribution.py

```diff
@@ -0,0 +1,738 @@
+            Probability of grass establishment (-)
+        sh_max_age: int, optional
+            Maximum age of shrubs (yr)
+        sh_seedling_max_age: int, optional
+            Maximum age of shrub seedlings (yr)
+        sh_seedling_mor_dis: float, optional
+            Probability of mortality of shrub seedlings due to disease (-)
+        sh_mor_dis_low_thresh_age: int, optional
+            Age at which shrubs start experiencing disease (yr)
+        sh_mor_dis_low_slp: float, optional
+            Probability of mortality of shrubs due to disease
+            at  sh_mor_dis_low_thresh_age (-)
+        sh_mor_dis_high_slp: float, optional
+            Probability of mortality of shrubs due to disease
+            at sh_max_age (-)
+        P_sh_fire_mor: float, optional
+            Probability of shrub mortality due to fire (-)
+        P_gr_fire_mor: float, optional
+            Probability of grass mortality due to fire (-)
+        sh_mor_ws_thresh: float, optional
+            Background mortality probability of shrubs due
+            to waterstress (P_mor_ws) (-)
+        gr_mor_ws_thresh: float, optional
+            Background mortality probability of grass due
+            to waterstress (-)
+        """
+
+        super(ResourceRedistribution, self).__init__(grid, **kwds)
+
+        name = "vegetation__plant_functional_type"
+        if name not in self.grid.at_cell:
+            print(
+                "Since a cellular field of PFTs "
+                + "is not provided, the field "
+                + name
+                + " is initialized to 0 - all bare cells!"
+            )
+            self.grid.add_zeros("cell", name, units=self._var_units[name])
+        name = "soil__resources"
+        if name not in self.grid.at_cell:
+            print(
+                "Since a cellular field for Resources "
+                + "is not provided, the field "
+                + name
+                + " is initialized to 1!"
+            )
+            self.grid.add_ones("cell", name, units=self._var_units[name])
+
+        self.initialize(
+            e=e,
+            R_low_threshold=R_low_threshold,
+            R_threshold=R_threshold,
+            R_dep_threshold=R_dep_threshold,
+            Rth_gr=Rth_gr,
+            Rth_sh=Rth_sh,
+            P_gr_regrwth=P_gr_regrwth,
+            P_sh_regrwth=P_sh_regrwth,
+            Pen=Pen,
+            Pgrz=Pgrz,
+            P_gr=P_gr,
+            sh_max_age=sh_max_age,
+            sh_seedling_max_age=sh_seedling_max_age,
+            sh_seedling_mor_dis=sh_seedling_mor_dis,
+            sh_mor_dis_low_thresh_age=sh_mor_dis_low_thresh_age,
+            sh_mor_dis_low_slp=sh_mor_dis_low_slp,
+            sh_mor_dis_high_slp=sh_mor_dis_high_slp,
+            P_sh_fire_mor=P_sh_fire_mor,
+            P_gr_fire_mor=P_gr_fire_mor,
+            sh_mor_ws_thresh=sh_mor_ws_thresh,
+            gr_mor_ws_thresh=gr_mor_ws_thresh,
+            **kwds,
+        )
+
+    def initialize(
```

Comment:
This method seems unnecessary? Why not just initialize these variables in *__init__*?

Reply:
This is intentional. The reason I have another method is that I would like to keep the option of re-initializing the inputs (probably a subset of them) and use the same object to do so.